File: cpp/CSharp/CPP14ParserBase.cs
using Antlr4.Runtime;
using System.IO;

public abstract class CPP14ParserBase : Parser {

    protected CPP14ParserBase(ITokenStream input, TextWriter output, TextWriter errorOutput)
        : base(input, output, errorOutput)
    {
    }

    public bool IsPureSpecifierAllowed()
    {
        try
        {
            var x = this.Context; // memberDeclarator
            var c = x.GetChild(0).GetChild(0);
            var c2 = c.GetChild(0);
            var p = c2.GetChild(1);
            if (p == null) return false;
            return p.GetType() == typeof(CPP14Parser.ParametersAndQualifiersContext);
        }
        catch
        {
        }
        return false;
    }
}

File: cpp/Dart/CPP14ParserBase.dart
import 'package:antlr4/antlr4.dart';
import 'dart:io';
import 'dart:convert';
import 'CPP14Parser.dart';

abstract class CPP14ParserBase extends Parser
{
    CPP14ParserBase(TokenStream input) : super(input)
    {
    }

    bool IsPureSpecifierAllowed()
    {
        try
        {
            var x = this.context; // memberDeclarator
            var c = x?.getChild(0)?.getChild(0);
            var c2 = c?.getChild(0);
            var p = c2?.getChild(1);
            if (p == null) return false;
            return p.runtimeType == ParametersAndQualifiersContext;
        }
        catch (e)
        {
        }
        return false;
    }
}
